gauntlet/packages/gauntlet-solana/src/commands/internal/solana.ts

```diff
@@ -63,6 +68,38 @@ export default abstract class SolanaCommand extends WriteCommand<TransactionResp
     }
   }
 
+  signAndSendRawTx = async (rawTxs: RawTransaction[], extraSigners?: Keypair[]): Promise<TransactionSignature> => {
+    const latestSlot = await this.provider.connection.getSlot()
+    const recentBlock = await this.provider.connection.getBlock(latestSlot)
+    const tx = makeTx(rawTxs, {
+      recentBlockhash: recentBlock.blockhash,
+      feePayer: this.wallet.publicKey,
+    })
+    if (extraSigners) {
+      tx.sign(...extraSigners)
+    }
+    const signedTx = await this.wallet.signTransaction(tx)
+    logger.loading('Sending tx...')
+    return await sendAndConfirmRawTransaction(this.provider.connection, signedTx.serialize())
+  }
+
+  withIDL = (action: (...args: any) => Promise<TransactionSignature>, idl: Idl) => async (
```

Comment:
Hmm, calling this method doesn't read very well to me:

```
const txhash = await this.withIDL(this.signAndSendRawTx, program.idl)(rawTxs, [feed])
```

Reply:
We always send the same `this.signAndSendRawTx` action in, so makes sense for this fn to be restructured slightly based on it's demonstrated usage.

Reply:
Hmm, I don't want to mess up `signAndSendRawTx` with contract related stuff, as is quite clear what it does. Changing the naming for `sendTxWithIDL` sounds better?